### DISCUSSION TITLE
Armbian 安装 Home Assistant Supervised 实践

### DISCUSSION THREAD

**Discussion by mjl371** [ophub/amlogic-s9xxx-armbian#2229] [Amlogic]
# Armbian 安装 Home Assistant Supervised 实践

## 前言(劝退)
先了解有没必要安装Supervised，Supervised/OS和Core/Container的主要区别在于前者可以安装Add-ons和进行备份还原，后者可以进行备份但不能还原。Core/Container已经可以满足绝大部分需求，可以安装HACS。Add-ons的大部分应用在宿主机上(Armbian)都可以单独安装使用。而安装了Supervised后，docker则不能额外启用其他容器，否则会有unhealthy警告，极大的降低了Armbian的可玩性。
如果只需要安装Container只需使用Armbian-software安装 Home Assistant

## 0.安装环境与准备工作

本文以设备H96-pro-plus为例，SoC是amlogic_s912
选择镜像`Armbian_bookworm`，即Debian12。[安装Supervised系统需求](https://github.com/home-assistant/architecture/blob/master/adr/0014-home-assistant-supervised.md)
Armbian[系统安装与启动](https://github.com/ophub/amlogic-s9xxx-armbian/blob/main/README.cn.md)
请准备良好的网络环境，国内更换镜像源能加快安装，但不能完全解决网络问题，安装Supervised与系统初始化时需要频繁访问互联网。
[官方安装流程](https://github.com/home-assistant/supervised-installer)
[参考教程:https://haade.fr/en/blog/easy-installation-home-assistant-os-armbian-cubietruck-2024](https://haade.fr/en/blog/easy-installation-home-assistant-os-armbian-cubietruck-2024)
- ### 更换 Armbian 源

登录 Armbian 系统，输入命令，按需选择源

```bash
armbian-apt
```

### 安装依赖

```bash
apt install \
apparmor \
cifs-utils \
curl \
dbus \
jq \
libglib2.0-bin \
lsb-release \
network-manager \
nfs-common \
systemd-journal-remote \
systemd-resolved \
udisks2 \
wget -y
```
## 2.添加设置
修改系统版本名`PRETTY_NAME`为`Debian GNU/Linux 12 (bookworm)`
```bash
nano /etc/os-release
```

```bash
PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
```

 [修改 cmdline 设置](https://github.com/ophub/amlogic-s9xxx-armbian/blob/main/documents/README.cn.md#1214-%E5%A6%82%E4%BD%95%E4%BF%AE%E6%94%B9-cmdline-%E8%AE%BE%E7%BD%AE) 不同SoC设备位置不同

```bash
nano /boot/uEnv.txt
```

行尾添加

设置`docker cgroup v1`版本
```bash
systemd.unified_cgroup_hierarchy=0
```

启用`apparmor`
```bash
extraargs=apparmor=1 security=apparmor
```

重启生效
```bash
reboot
```

## 3.安装docker

```bash
armbian-software
```
选择docker的ID

或使用安装脚本
```bash
curl -fsSL get.docker.com | sh
```

## 4.安装[OS-Agent](https://github.com/home-assistant/os-agent/releases)
选择合适的架构和版本，以aarch64和1.6.0为例

```bash
wget -O os-agent.deb https://github.com/home-assistant/os-agent/releases/download/1.6.0/os-agent_1.6.0_linux_aarch64.deb
apt install ./os-agent.deb
```
检查状态
```bash
gdbus introspect --system --dest io.hass.os --object-path /io/hass/os
```
## 5.安装supervised installer
前面都是准备工作，这个才是官方的supervised安装脚本
```bash
wget -O homeassistant-supervised.deb https://github.com/home-assistant/supervised-installer/releases/latest/download/homeassistant-supervised.deb
apt install ./homeassistant-supervised.deb
```

图形界面选择合适的架构

```bash
qemuarm64
```
等待安装完成，需要初始化较长时间

**Comment by ophub** [comment]
写的挺详细，点个赞

**Reply by 475617747** [reply]
改完了还是显示不支持，不知道咋回事
error] Armbian-unofficial 24.5.0-trunk noble is not supported!
dpkg: error processing archive /root/homeassistant-supervised.deb (--unpack):new homeassistant-supervised package pre-installation script subprocess returned errorexit status 1info] Undo divert on abort-installNo diversion 'diversion of /etc/NetworkManager/NetworkManager.conf to /etc/NetworkManacer/NetworkManager .conf.real by homeassistant-supervised' , none removed.No diversion 'diversion of /etc/NetworkManager/system-connections/default to /etc/NetworkManager/system-connections/default,real by homeassistant-supervised', none removed.No diversion 'diversion of /etc/docker/daemon.json to /etc/docker/daemon.json.real by homeassistant-supervised', none removed.No diversion 'diversion of /etc/network/interfaces to /etc/network/interfaces.real by homeassistant-supervised',none removed.info] Resetting debconfselectionsdue to the complexity of this installation method,info]Removal complete,info]you will need to manually remove the containers created by the supervisorErrors were encountered while processing:/root/homeassistant-supervised.debN: Download is performed unsandboxed as root as file '/root/homeassistant-supervised.decouldn't be accessed by user ' apt'.- pkgAcquire::Run (13: Permission denied)E: Sub-process /usr/bin/dpkg returned an error code (1)

**Reply by mjl371** [reply]
> 改完了还是显示不支持，不知道咋回事
> 
> error] Armbian-unofficial 24.5.0-trunk noble is not supported!
> 
> dpkg: error processing archive /root/homeassistant-supervised.deb (--unpack):new homeassistant-supervised package pre-installation script subprocess returned errorexit status 1info] Undo divert on abort-installNo diversion 'diversion of /etc/NetworkManager/NetworkManager.conf to /etc/NetworkManacer/NetworkManager .conf.real by homeassistant-supervised' , none removed.No diversion 'diversion of /etc/NetworkManager/system-connections/default to /etc/NetworkManager/system-connections/default,real by homeassistant-supervised', none removed.No diversion 'diversion of /etc/docker/daemon.json to /etc/docker/daemon.json.real by homeassistant-supervised', none removed.No diversion 'diversion of /etc/network/interfaces to /etc/network/interfaces.real by homeassistant-supervised',none removed.info] Resetting debconfselectionsdue to the complexity of this installation method,info]Removal complete,info]you will need to manually remove the containers created by the supervisorErrors were encountered while processing:/root/homeassistant-supervised.debN: Download is performed unsandboxed as root as file '/root/homeassistant-supervised.decouldn't be accessed by user ' apt'.- pkgAcquire::Run (13: Permission denied)E: Sub-process /usr/bin/dpkg returned an error code (1)

error] Armbian-unofficial 24.5.0-trunk noble is not supported!
第一句就说明镜像不对啊,要bookworm的

**Comment by liuxing-maker** [comment]
请教一下，如果用apt install ./homeassistant-supervised.deb安装就会提示权限问题，如下图：
![1](https://github.com/user-attachments/assets/1727b68d-8b8b-428b-a6b0-c436557f2be9)
如果用指令dpkg -i homeassistant-supervised.deb安装，不会报错，但是服务好像没启动，浏览器输入地址无法访问，如下图：
![22](https://github.com/user-attachments/assets/c1b08beb-3540-4373-b68e-c39baab7e747)
![3](https://github.com/user-attachments/assets/477fa7c5-4749-4f13-9da2-747b3d5050d1)

**Reply by mr-jia1994** [reply]
我也是用dpkg。等一会就有了，在下载。
apt的话需要把homeassistant-supervised.deb  放到其他 如/mnt目录下，给个777权限也能装

**Comment by noctiro** [comment]
要改的是 `/usr/lib/os-release` , 而不是 `/etc/os-release`

修改系统版本名`PRETTY_NAME`为`Debian GNU/Linux 12 (bookworm)` (11在`2.0.0`版本停止支持)

**Comment by hhyygg2009** [comment]
可以这样不修改系统名称 
`sudo BYPASS_OS_CHECK=true apt install ./homeassistant-supervised.deb`

**Reply by bingshui20** [reply]
> 可以这样不修改系统名称 `sudo BYPASS_OS_CHECK=true apt install ./homeassistant-supervised.deb`

<img width="1280" height="739" alt="b7ec01759ec15e1c879e6bd2d2e437ea_720" src="https://github.com/user-attachments/assets/47acaa81-3b16-444f-a824-73363293fe65" />
这样是安装完成了吗
